pyproject.toml

```diff
@@ -66,6 +66,7 @@ strict_optional = true
 warn_redundant_casts = true
 warn_unused_configs = true
 warn_unused_ignores = true
+exclude = ["tests/"]
```

Comment:
Why are we introducing this exclude line here? I am struggling to understand why it is relevant to the rest of the PR

Reply:
ups